src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -322,49 +322,45 @@ Klass* SystemDictionary::resolve_array_class_or_null(Symbol* class_name,
   return k;
 }
 
+static inline void log_circularity_error(PlaceholderEntry* probe) {
+  LogTarget(Debug, class, load, placeholders) lt;
+  if (lt.is_enabled()) {
+    ResourceMark rm;
+    LogStream ls(lt);
+    ls.print("ClassCircularityError detected for placeholder ");
+    probe->print_entry(&ls);
+    ls.cr();
+  }
+}
 
 // Must be called for any super-class or super-interface resolution
 // during class definition to allow class circularity checking
 // super-interface callers:
-//    parse_interfaces - for defineClass
+//    parse_interfaces - from defineClass
 // super-class callers:
-//   ClassFileParser - for defineClass
+//   ClassFileParser - from defineClass
```

Comment:
picky review comment for line #340, comments below do not use a hyphenated "super-class", but instead one work "superclass".

Reply:
Ok.  How about a space?  Because the sentence has super-class and super-interface, so I want to change them both.

Reply:
Lois pointed out that the spec uses superclass and superinterface, so I changed the comments to use this convention.